File: front/picts_manager/lib/model/core/Album.dart
import 'package:picts_manager/model/core/User.dart';

class Album {
  String albumId;

  String name;

  String idOwner;

  bool isPublic;

  String? imageId;

  Album({
    required this.albumId,
    required this.name,
    required this.idOwner,
    required this.isPublic,
    this.imageId,
  });

  factory Album.fromJson(Map<String, dynamic> json) {
    try {
      return Album(
          albumId: json['album']["albumId"].toString(),
          name: json['album']["albumName"].toString(),
          idOwner: json['album']["albumIdOwner"].toString(),
          isPublic: json['album']["albumIsPublic"].toString() == "1",
          imageId: json['image']['imageId'].toString());
    } catch (e) {
      try {
        return Album(
          albumId: json['album']["albumId"].toString(),
          name: json['album']["albumName"].toString(),
          idOwner: json['album']["albumIdOwner"].toString(),
          isPublic: json['album']["albumIsPublic"].toString() == "1",
        );
      } catch (e) {
        return Album(
          albumId: json["albumId"].toString(),
          name: json["albumName"].toString(),
          idOwner: json["albumIdOwner"].toString(),
          isPublic: json["albumIsPublic"].toString() == "1",
        );
      }
    }
  }

  factory Album.fromJson2(Map<String, dynamic> json) {
    return Album(
        albumId: json["albumId"].toString(),
        name: json["albumName"].toString(),
        idOwner: json["albumIdOwner"].toString(),
        isPublic: json["albumIsPublic"].toString() == "1",
        imageId: json['imageId'].toString());
  }
}

File: back/src/main/java/com/epitech/pictsmanager/model/Album.java
package com.epitech.pictsmanager.model;

import jakarta.persistence.*;
import java.util.Set;

@Entity
@Table(name="albums")
public class Album {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer albumId;

    @Column(name = "name", nullable = false, length = 120)
    private String albumName;

    @Column(name = "is_public", nullable = false)
    private int albumIsPublic;

    @Column(name = "is_default", nullable = false)
    private int albumIsDefault;

    @Column(name = "id_owner", nullable = false)
    private int albumIdOwner;

    @OneToMany(mappedBy = "album", cascade = CascadeType.ALL)
    private Set<Image> imageSet;

    @OneToMany(mappedBy = "albumUser", cascade = CascadeType.ALL)
    private Set<AlbumShared> albumSharedUserSet;

    public Integer getAlbumId() {
        return albumId;
    }

    public void setAlbumId(Integer albumId) {
        this.albumId = albumId;
    }

    public String getAlbumName() {
        return albumName;
    }

    public void setAlbumName(String albumName) {
        this.albumName = albumName;
    }

    public int getAlbumIsPublic() {
        return albumIsPublic;
    }

    public void setAlbumIsPublic(int albumIsPublic) {
        this.albumIsPublic = albumIsPublic;
    }

    public int getAlbumIdOwner() {
        return albumIdOwner;
    }

    public void setAlbumIdOwner(int albumIdOwner) {
        this.albumIdOwner = albumIdOwner;
    }

    public int getAlbumIsDefault() {
        return albumIsDefault;
    }

    public void setAlbumIsDefault(int albumIsDefault) {
        this.albumIsDefault = albumIsDefault;
    }

}
